examples/src/bin/custom_editable.rs

```diff
@@ -0,0 +1,376 @@
+                        "label",
+                        "Label",
+                        "Label",
+                        Some(""),
+                        glib::ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::boolean(
+                        "has-close-button",
+                        "Has close button",
+                        "Whether this tag has a close button",
+                        false,
+                        glib::ParamFlags::READWRITE,
+                    ),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn get_property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> glib::Value {
+            match pspec.get_name() {
+                "label" => self.label.get_text().to_value(),
+                "has-close-button" => self.has_close_button.get().to_value(),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn set_property(
+            &self,
+            tag: &Self::Type,
+            _id: usize,
+            value: &glib::Value,
+            pspec: &ParamSpec,
+        ) {
+            match pspec.get_name() {
+                "label" => self.label.set_text(value.get().unwrap().unwrap()),
+                "has-close-button" => {
+                    tag.set_has_close_button(value.get_some().unwrap());
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        fn signals() -> &'static [Signal] {
+            use once_cell::sync::Lazy;
+            static SIGNALS: Lazy<Vec<Signal>> = Lazy::new(|| {
+                vec![
+                    Signal::builder("closed", &[], <()>::static_type().into()).build(),
+                    Signal::builder("clicked", &[], <()>::static_type().into()).build(),
+                ]
+            });
+            SIGNALS.as_ref()
+        }
+
+        fn constructed(&self, tag: &Self::Type) {
+            self.container.set_parent(tag);
+            self.container.append(&self.label);
+
+            let gesture = gtk::GestureClick::new();
+            gesture.connect_released(clone!(@weak tag => move |_gesture, _n_press, _x, _y| {
+                tag.emit(CustomTag::signals()[1].signal_id(), &[]).unwrap();
+            }));
+            tag.add_controller(&gesture);
+        }
+
+        fn dispose(&self, _tag: &Self::Type) {
+            self.container.unparent();
+        }
+    }
+    impl WidgetImpl for CustomTag {
+        fn measure(
+            &self,
+            _widget: &Self::Type,
+            orientation: gtk::Orientation,
+            for_size: i32,
+            min: &mut i32,
+            nat: &mut i32,
+            min_base: &mut i32,
+            nat_base: &mut i32,
+        ) {
+            let (c_min, c_nat, c_min_base, c_nat_base) =
+                self.container.measure(orientation, for_size);
+            *min = c_min;
+            *nat = c_nat;
+            *min_base = c_min_base;
+            *nat_base = c_nat_base;
```

Comment:
This is definitely not very nice I think; we should probably just return `(i32, i32, i32, i32)` with the subclassing code.

Reply:
I guess so :) But a) do those 4 pointers contain values we might have to read, b) do all 4 always have to be returned?

Reply:
I will leave this for another time and opened to not forget about it https://github.com/gtk-rs/gtk4-rs/issues/222